src/ovirtapi.js

```diff
@@ -839,6 +867,19 @@ OvirtApi = {
     OvirtApi._assertLogin({ methodName: 'getNetworks' })
     return OvirtApi._httpGet({ url: `${AppConfiguration.applicationContext}/api/networks` })
   },
+  addHttpListener (listener: ListenerType) {
+    listeners.add(listener)
+  },
+}
+
+function notifyStart (method: MethodType, url: string): Object {
+  const requestId = { method, url }
```

Comment:
~~Will this be unique enough?~~  It should be since you're using a Set from ImmutableJS in the reducer to keep track.

If 2 requests with the same URL get fired right after each other (or concurrently), the requestId would be removed on completion of the first request.  If the second doesn't complete within 500ms the "Loading..." indicator would go away while the request is still in progress.  That's a long interval, and I can't think of a particular scenario where it would happen off the top of my head.

I guess the question is, would it be acceptable to have the "Loading..." indicator disappear in an edge case while a second request is long running (over 500ms)?

Reply:
If already refactoring, I would make it reliable. The `requestId` should be unique for each HTTP call. Please see comment above, simple global counter can ensure that.

I would not re-compose the requestId here again since it's already passed at calling (so allign it with the `notifyStop()`)

Reply:
* The requestId properties `method` and `url` are there for debugging purposes only. The notify methods as well as immutable.js Set works with object identity (`===`) so the uniqueness is inherently ensured (Items in the Set are pure JS objects, not immutable.js instances). In another words it would work with empty objects (`{}`) as well. Please let me know in case the `requestId` serial number is required even for debugging purposes only.
* Regarding place of creation of requestId: It's created in `notifyStart()` since content of the requestId should be transparent to the calling party. The contract is supposed to be like this: Call `notifyStart()`, save the token (requestId) and whenever appropriate cancel that notification by passing the token to `notifyStop()`.
* It actually works - multiple concurrent requests in using the same method to the same url creates multiple records in the `/activeRequests` Set.

Reply:
@jniederm , you are right, I oversaw that.